SonarRunner.Shim/SonarRunner.Wrapper.cs

```diff
@@ -101,6 +115,26 @@ private static bool ExecuteJavaRunner(ILogger logger, string exeFileName, string
             return success;
         }
 
+        /// <summary>
+        /// Get the value of the SONAR_RUNNER_OPTS variable that controls the amount of memory available to the JDK so that the sonar-runner doesn't 
+        /// hit OutOfMemory exceptions. If no env variable with this name is defined at machine or user level then use the process one is used. 
+        /// Bar that, a default value is used. 
+        /// </summary>
+        /// <returns></returns>
+        private static string GetSonarOptsValue()
+        {
+            if (!String.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable(SonarOptsVariable, EnvironmentVariableTarget.Machine)) ||
```

Comment:
Is it necessary to check the Machine, User and Process variables separately? The documentation for the GetEnvironmentVariable(string) overload says it takes into account the per-machine, per-user and per-process variables: https://msdn.microsoft.com/en-us/library/77zkk0b6(v=vs.110).aspx.